dotnet/src/Skills/Skills.OpenAPI/RestApiOperationRunner.cs

```diff
@@ -147,21 +189,117 @@ public RestApiOperationRunner(HttpClient httpClient, AuthenticateRequestAsyncCal
             }
         }
 
-        if (!s_payloadFactoryByMediaType.TryGetValue(mediaType!, out var payloadFactory))
+        if (!this._payloadFactoryByMediaType.TryGetValue(mediaType!, out var payloadFactory))
         {
             throw new SKException($"The media type {mediaType} of the {operation.Id} operation is not supported by {nameof(RestApiOperationRunner)}.");
         }
 
-        return payloadFactory.Invoke(arguments);
+        return payloadFactory.Invoke(operation.Payload, arguments);
     }
 
     /// <summary>
     /// Builds "application/json" payload.
     /// </summary>
+    /// <param name="payloadMetadata">The payload meta-data.</param>
     /// <param name="arguments">The payload arguments.</param>
     /// <returns>The HttpContent representing the payload.</returns>
-    private static HttpContent BuildAppJsonPayload(IDictionary<string, string> arguments)
+    private HttpContent BuildAppJsonPayload(RestApiOperationPayload? payloadMetadata, IDictionary<string, string> arguments)
```

Comment:
I would split the logic of this method in a couple of smaller methods to simplify code reading and maintenance.

Reply:
Done.